sense-android-library/src/nl/sense_os/datastorageengine/RealmModelSensor.java

```diff
@@ -0,0 +1,232 @@
+package nl.sense_os.datastorageengine;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import io.realm.Realm;
+import io.realm.RealmObject;
+import io.realm.annotations.Index;
+import io.realm.annotations.PrimaryKey;
+import nl.sense_os.service.shared.SensorDataPoint.DataType;
+
+public class RealmModelSensor extends RealmObject {
+
+    // compoundKey is used purely for Realm, to prevent duplicate sensors.
+    // it's built up from the sensors `source` and `name`.
+    @PrimaryKey
+    private String compoundKey = null;
```

Comment:
@josdejong in iOS, primaryKey is id for Sensor, and the combination of source and sensor name is checked when the sensor is about to be created. It is fine, but I feel a bit weird since id here is used as a foreign key in dataPoint class. Let's discuss about this.